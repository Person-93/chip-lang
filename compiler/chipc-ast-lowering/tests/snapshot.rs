use chip_snapshot_tests::{assert_snapshot, cases, run_case};
use chipc_ast::Ast;
use chipc_ast_lowering::lower_package;
use chipc_hir::HirPretty;

#[chip_test_harness::nextest_main(snapshot)]
fn main(name: &str) {
  let source = run_case(name);
  let ast = Ast::parse(&source);
  let hir = lower_package(&ast);
  let hir = format!("{}", hir.root.pretty_print(&hir).format());
  let hir = if hir.is_empty() {
    "// empty hir".to_string()
  } else {
    hir
  };
  assert_snapshot!(hir);
}
