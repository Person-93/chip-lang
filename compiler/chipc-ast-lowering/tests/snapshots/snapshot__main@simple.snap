---
source: compiler/chipc-ast-lowering/tests/snapshot.rs
assertion_line: 17
expression: hir
---
priv fn f() -> ()
  ()
